### PR TITLE
Docs Update and New `processing_lvl` implementation

### PR DESCRIPTION
**Pull request by YiweiMao**
- collapsing sidebar issue fixed (custom sidebar.json file. no longer relying on auto-generated one)
- made placeholder geometry.html page and included in sidebar
- quick start guide made
- added import instructions to each docs page
- added some function annotations and docstrings
- tidied up some naming conventions (noframe -> nframes), getNimgs -> avgNimgs, specta_pt_contoller -> SpectraPTController
- new `processing_lvl` implementation to make it easier to understand. Mainly for the Transforms Setup method. Core strategy stays the same.
- fixed bug relating to CameraProperties.slow_binning
- fixed bug relating to switching the order of dn2rad in pipeline
- all processing levels now work? At least on the FLIR camera and a dodgy self generated integrating sphere datacube
- update_intsphere_cube outputs a 4D cube (previously gave a 5D cube with one axis length 1, harmless np.squeeze left in code)

Update:
- fixed edge case with fit_HgAr_lines
- implemented ProcessRawDatacube
- cleaned up set_processing_lvl some more

### CODE REVIEW

**Review by christhechris — APPROVED** [review]
looks pretty sensible to me, much clearer pathways now for processing levels, good idea.

File: openhsi/data.py [hunk 1]
```diff
@@ -277,10 +275,10 @@ def rad2ref_6SV(self:CameraProperties, x:Array['λ,x',np.float32]) -> Array['λ,
 @patch
 def set_processing_lvl(self:CameraProperties, lvl:int = 2, custom_tfms:List[Callable[[np.ndarray],np.ndarray]] = None):
```
Comment (christhechris) [inline openhsi/data.py:276]: just noticed the doc string says -1 is default, but 2 is default in func def.

**Review by christhechris — APPROVED** [review]

File: openhsi/data.py (hunk 1 above)
Comment (YiweiMao) [reply openhsi/data.py:276]: nice catch. I got it in the docs, the tutorial, the docstring, in the __init__ but not the func def itself! 
fixed in upcoming commit.

**Comment by YiweiMao** [comment]
- refactored `DataCube.load_nc` and `ProcessRawDatacube`
- saving *after* further processing retains the camera temps and capture times in the original file (discards the original metadata though... will have to work on that)
I don't see any issue with merging changes so I'm going to go ahead. 
You'll want to update example.ipynb with the changes I've made to ProcessDatacube in notebook 01.

**Pull request merged**